app/components/AddTodoCard.tsx

```diff
@@ -1,15 +1,30 @@
 // components/AddTodoCard.tsx
 import { PlusIcon } from "@heroicons/react/20/solid";
-import React from "react";
+import React, { useEffect, useState } from "react";
 import { useForm } from "react-hook-form";
-import { TodoFormInputs, ITodo } from "@/types";
+import { TodoFormInputs, ITodo, IUser } from "@/types";
 import { v4 as uuidv4 } from "uuid";
+import dataService from "@/services/dataService";
 
 export default function AddTodoCard({
   onAdd,
 }: {
   onAdd: (item: ITodo) => void;
 }) {
+  const [userData, setUserData] = useState<IUser[] | null>(null);
+
+  useEffect(() => {
```

Comment:
normally, data fetching should be done in higher order component, for example, in page